src/main/java/com/redhat/devtools/lsp4ij/features/documentation/SyntaxColorationCodeBlockRenderer.java

```diff
@@ -0,0 +1,156 @@
+            myEndOffset = endOffset;
+            myIterator = highlighter.createIterator(startOffset);
+        }
+
+        @Override
+        public boolean atEnd() {
+            return myIterator.atEnd() || getCurrentStart() >= myEndOffset;
+        }
+
+        private int getCurrentStart() {
+            return Math.max(myIterator.getStart(), myStartOffset);
+        }
+
+        private int getCurrentEnd() {
+            return Math.min(myIterator.getEnd(), myEndOffset);
+        }
+
+        @Override
+        public void advance() {
+            int prevEnd = myCurrentEnd;
+            myCurrentStart = getCurrentStart();
+            myCurrentEnd = getCurrentEnd();
+            assert prevEnd <= myCurrentStart && myCurrentStart <= myCurrentEnd
+                    : "Unexpected range returned by highlighter: " +
+                    myIterator.getStart() + ":" + myIterator.getEnd() +
+                    ", prevEnd: " + prevEnd +
+                    ", scanned range: " + myStartOffset + ":" + myEndOffset +
+                    ", resulting range: " + myCurrentStart + ":" + myCurrentEnd +
+                    ", highlighter: " + myHighlighter;
+            myCurrentAttributes = myIterator.getTokenType() == TokenType.BAD_CHARACTER ? EMPTY_ATTRIBUTES : myIterator.getTextAttributes();
+            myIterator.advance();
+        }
+
+        @Override
+        public int getRangeStart() {
+            return myCurrentStart;
+        }
+
+        @Override
+        public int getRangeEnd() {
+            return myCurrentEnd;
+        }
+
+        @Override
+        public TextAttributes getTextAttributes() {
+            return myCurrentAttributes;
+        }
+
+        @Override
+        public void dispose() {
+        }
+
+        @Override
+        public String toString() {
+            return "HighlighterRangeIterator[" + myHighlighter + "]";
+        }
+    }
+
+    private void render(FencedCodeBlock node, NodeRendererContext context, HtmlWriter html) {
+        String language = node.getInfo().toString();
+        CharSequence code = node.getContentChars();
+        html.line();
+
+        Language lang = LightQuickDocHighlightingHelper.guessLanguage(language);
+        if (lang != null) {
+            // Case 1: Highlight code block with the guessed language
+            String s = LightQuickDocHighlightingHelper.getStyledSignatureFragment(project, lang, node.getContentChars().toString());
+            html.append(s);
+        } else {
+            // Case 2: Try to highlight code block with TextMate
+            if (!(HAS_TEXTMATE_SUPPORT && TextMateHighlighterHelper.highlightWithTextMate(code, language, html))) {
+                // Case 3 : no highlight
+                html.srcPos(node.getChars()).withAttr().tag("pre").openPre();
+                html.withAttr().tag("code");
+                html.text(node.getContentChars());
+                html.tag("/code").closeTag("pre").closePre();
+            }

```

Comment:
```java
html.srcPos(node.getChars()).withAttr().tag("pre").openPre();
                if (!(HAS_TEXTMATE_SUPPORT && TextMateHighlighterHelper.highlightWithTextMate(code, language, html))) {
      html.withAttr().tag("code");
      html.text(node.getContentChars());
      html.tag("/code");
  }
  html.closeTag("pre").closePre();
  ```
  to get that monospace font on code blocks

Reply:
I wrap now with pre.